assembl/static2/js/app/components/common/termsForm.jsx

```diff
@@ -0,0 +1,113 @@
+// @flow
+import React from 'react';
+import { Translate } from 'react-redux-i18n';
+import { Modal, Button } from 'react-bootstrap';
+import { compose, graphql } from 'react-apollo';
+import type { OperationComponent, QueryProps } from 'react-apollo';
+import { closeModal } from '../../utils/utilityManager';
+import LegalNoticeAndTerms from '../../graphql/LegalNoticeAndTerms.graphql';
+import withLoadingIndicator from '../../components/common/withLoadingIndicator';
+
+type TermsFormProps = {
+  isChecked: boolean,
+  text: string,
+  handleAcceptButton: () => void
+};
+
+type TermsFormState = {
+  isScrolled: boolean
+};
+
+class DumbTermsForm extends React.Component<*, TermsFormProps, TermsFormState> {
+  props: TermsFormProps;
+
+  state: TermsFormState;
+
+  box: HTMLElement;
+
+  handleSubmit: () => void;
+
+  constructor() {
+    super();
+    this.state = {
+      isScrolled: false
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.box.addEventListener('scroll', this.trackScrolling);
+  }
+
+  trackScrolling = (): void => {
+    const wrappedElement = this.box;
+    if (wrappedElement.scrollHeight - wrappedElement.scrollTop === wrappedElement.clientHeight) {
+      this.setState({
+        isScrolled: true
+      });
+      wrappedElement.removeEventListener('scroll', this.trackScrolling);
+    }
+  };
+
+  handleSubmit() {
+    this.props.handleAcceptButton();
+    closeModal();
+  }
+
+  render() {
+    const { isScrolled } = this.state;
+    const { isChecked, text } = this.props;
+    return (
+      <div className="terms-form">
+        <Modal.Header closeButton>
+          <Modal.Title>
+            <Translate value="termsAndConditions.headerTitle" />
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <div
+            className="terms-box justify"
+            ref={(box) => {
+              this.box = box;
+            }}
+            dangerouslySetInnerHTML={{ __html: text }}
+          />
+          {isScrolled &&
+            !isChecked && (
```

Comment:
I don't understand why you did such a complex thing with the scroll event. For me there is no need of this `isScrolled`. The button is always there if `!isChecked`, the user see the button once he scrolled down the text.

Reply:
voir tests d'acceptance

Reply:
I just follow the acceptance tests

Reply:
I read them, but for me it's still valid if you always show the button. Do you really notice a difference between rendering the button when you scroll to the bottom and always rendering it?

Reply:
ok will change it tomorrow